content/spin/v2/key-value-store-tutorial.md

```diff
@@ -522,6 +522,6 @@ Congratulations, you have a Spin application and associated Key Value store runn
 
 ## Next Steps
 
-* Explore the contents of your Key Value store with the [Key Value Store Explorer template](https://developer.fermyon.com/hub/preview/template_kv_explorer)
+* Explore the contents of your Key Value store with the [Key Value Store Explorer template](../../api/hub/preview/template_kv_explorer)
```

Comment:
This differs from the previous path by adding `api` on the front.  For me locally this causes it to 404.

Reply:
I am curious why the link checker does not pick up on this one...

Reply:
Yeah, the `api/hub` baffle me.
I think we can't link to the home page of the hub relatively. It will be the one exception. 
I have made this one absolute to get around the 404 issue.
Perhaps @karthik2804 will know why we are not able to link to `api/hub` index page using relative path.

Reply:
There are 2 sets of pages - one is `api/hub/...` which serves JSON data for the front-end (this should not be linked anywhere or site mapped). The other is what is served from the front-end which is `hub/...`. Therefore the URL here needs to be `/hub/preview/template_kv/explorer` or if you want relative URL, it is `../../hub/preview/template_kv_explorer`. 

The preview page comes from the `vue` app and is not part of the content files.

Reply:
Thanks, Karthik,
So if I understand it correctly there is no way to link to the https://developer.fermyon.com/hub (home page of the hub) relatively from within the GitHub markdown files. We will have to leave these `hub` related links as absolute links (I think there is only about 3 or 4 of them anyway).
This is ready for approval if you agree @itowlson - thanks so much!

Reply:
I am confused.  `../../hub/preview/template_kv_explorer` works for me (that is, it brings up the KV explorer Hub page), and seems to be what Karthik is suggesting.  Are you seeing something different @tpmccallum?

Reply:
Yeah, the issue is we can't link to the hub's landing page using relative links.
For example, there are instances in the docs where the hub's home page < https://github.com/fermyon/developer/pull/1323/files#diff-5c25aabf95d9789251ab01457e99ea55bf967e0d7e716f69208a64aa2817b61bL181 > is absolute and I wish we could link to the hub's home page using a relative link.

Reply:
There are only 3 links to the top-level landing page of the hub.
![image](https://github.com/fermyon/developer/assets/9831342/e77b9cd8-828f-4c61-8067-6fb3714b5852)
So don't want this to block the work. But it would be great to have that relative link ability, like we do with the other areas of the site.

Reply:
Oooh, might have found a loophole. 
If I use `/index` at the end of a relative link to the hub (i.e. `../hub/index`), it routes back to `/hub` (the user gets what they want), and the local broken link tests (npm run test) pass. I will try and push these changes and see if the build passes.

Reply:
> Linking to actual items is ok; the landing page is the issue.

I am still confused: don't we _want_ to link to an individual item here?  Sorry for being slow.

Reply:
All good Ivan, I will try and explain as best I can ...

Why link to the Spin Up Hub landing page?
The Python page, for example, is trying to link to the Spin Up Hub (in general) so users can dive in and use the filters to search for Python-related items. 

Why not just link to the Python section of the Spin Up Hub? 
When using the filters to browse different languages etc. the Spin Up Hub URL stays exactly the same. So we can never link to just Python-related things in the hub.

What types of links can we offer to users then?
There are only two types:
- A niche/specific link that only shows a single item i.e. [the Python HTTP Template](https://developer.fermyon.com/hub/preview/template_http_python) 
- [the landing page](https://developer.fermyon.com/hub)
There is no in between.

What was the original issue?
Sometimes (as is the case with this Python page) we just want to mention that a user can go to the Spin Up Hub at < https://developer.fermyon.com/hub > and use the filters to search for things. The Spin Up Hub works in such a way that using relative links in the documentation markdown files i.e. `../hub` did not pass the broken link checker in the build.

How has that been resolved?
It turns out that using `../hub/index` instead of just `../hub` manages to get past the broken link checker issue. We can now have 100% relative links in the documentation. 

Hope this makes sense.